### PR TITLE
PN-9549 - Navigation inside PP and ToS page

### PR DESCRIPTION
**Pull request by alessandrogelmi**
## Short description
Fix navigation inside PP and ToS page

## List of changes proposed in this pull request
- Used a function instead of hook to rewrite links

## How to test
On PF, PG and PA try to go to ToS or PP page and navigate the page through the links on the left

### CODE REVIEW

**Review by AndreaCimini90 — DISMISSED** [review]

File: packages/pn-pa-webapp/src/pages/PrivacyPolicy.page.tsx [hunk 1]
```diff
@@ -33,11 +31,9 @@ const PrivacyPolicyPage = () => {
   }, []);
 
   void waitForElement('.otnotice-content').then(() => {
```
Comment (AndreaCimini90) [inline packages/pn-pa-webapp/src/pages/PrivacyPolicy.page.tsx:33]: we can move this into the `OneTrust.NoticeApi.Initialized.then` callback. Do for all the apps

File: packages/pn-commons/src/utility/dom.utility.ts [hunk 2]
```diff
@@ -12,22 +12,12 @@ export enum WaitForElementResult {
  * // waits for myQuerySelector to be injected in the DOM.
  * waitForElem('.myQuerySelector').then(() => successCbk());
  */
-export function waitForElement(selector: string) {
-  return new Promise((resolve) => {
```
Comment (AndreaCimini90) [inline packages/pn-commons/src/utility/dom.utility.ts:16]: Maybe we can restore the old version. I strongly belive that the strict mode causes the malfunction in local environment

File: packages/pn-pa-webapp/src/main.tsx [hunk 3]
```diff
@@ -32,16 +32,16 @@ async function doTheRender() {
...
-          <BrowserRouter>
-            <ThemeProvider theme={theme}>
-              <CssBaseline />
-              <Suspense fallback="loading...">
-                <App />
-              </Suspense>
-            </ThemeProvider>
-          </BrowserRouter>
-        </React.StrictMode>
+        {/* <React.StrictMode> */}
```
Comment (AndreaCimini90) [inline packages/pn-pa-webapp/src/main.tsx:35]: Leave a comment to explain why we commented this code.
Do for all webapp and add reference to the current issue

File: packages/pn-pa-webapp/src/pages/PrivacyPolicy.page.tsx [hunk 4]
```diff
@@ -29,15 +27,13 @@ const PrivacyPolicyPage = () => {
           false
         );
       });
+
+      void waitForElement('.otnotice-content').then(() => {
```
Comment (AndreaCimini90) [inline packages/pn-pa-webapp/src/pages/PrivacyPolicy.page.tsx:31]: move into the then callback

File: packages/pn-pa-webapp/src/pages/TermsOfService.page.tsx [hunk 5]
```diff
@@ -24,19 +23,16 @@ const TermsOfServicePage = () => {
...
-              ONE_TRUST_DRAFT_MODE
-            ),
-          ],
+          [compileOneTrustPath(ONE_TRUST_TOS, ONE_TRUST_DRAFT_MODE)],
           false
         );
-        setContentLoaded(true);
+      });
+
+      void waitForElement('.otnotice-content').then(() => {
```
Comment (AndreaCimini90) [inline packages/pn-pa-webapp/src/pages/TermsOfService.page.tsx:31]: Same as before

File: packages/pn-personafisica-webapp/src/pages/TermsOfService.page.tsx [hunk 6]
```diff
@@ -14,26 +15,24 @@ declare const OneTrust: {
 };
 
 const TermsOfServicePage = () => {
-  const [contentLoaded, setContentLoaded] = useState(false);
-  const { ONE_TRUST_DRAFT_MODE, ONE_TRUST_TOS} = getConfiguration();
+  const configuration = useMemo(() => getConfiguration(), []);
```
Comment (AndreaCimini90) [inline packages/pn-personafisica-webapp/src/pages/TermsOfService.page.tsx:18]: Why the useMemo?

**Review by AndreaCimini90 — CHANGES_REQUESTED** [review]

File: packages/pn-personafisica-webapp/src/pages/TermsOfService.page.tsx (hunk 6 above)
Comment (alessandrogelmi) [reply packages/pn-personafisica-webapp/src/pages/TermsOfService.page.tsx:18]: removed also from pa

**Review by AndreaCimini90 — APPROVED** [review]

**Pull request merged**